lib/AST/ASTDemangler.cpp

```diff
@@ -1321,39 +1340,43 @@ ASTBuilder::findDeclContext(NodePointer node) {
       // If the generic signature is equivalent to that of the nominal type,
       // and we're in the same module, it's due to inverse requirements.
       // Just return the nominal declaration.
-      if (genericSigMatchesNominal &&
-          nominalDecl->getParentModule() == moduleDecl) {
-        return nominalDecl;
+      for (auto *moduleDecl : moduleDecls) {
+        if (genericSigMatchesNominal &&
+            nominalDecl->getParentModule() == moduleDecl) {
+          return nominalDecl;
+        }
       }
     }
 
-    for (auto *ext : nominalDecl->getExtensions()) {
-      if (ext->getParentModule() != moduleDecl)
-        continue;
-
-      if (!ext->isConstrainedExtension()) {
-        if (!genericSig || genericSigMatchesNominal)
-          return ext;
+    for (auto *moduleDecl : moduleDecls) {
+      for (auto *ext : nominalDecl->getExtensions()) {
```

Comment:
Again let's avoid a double nested loop here by changing the `if (ext->getParentModule() != moduleDecl)` to an `if (std::find(...))` or similar, and keeping the rest of the identation as-is